clang/test/SemaHLSL/resource_binding_attr_error.hlsl

```diff
@@ -1,9 +1,10 @@
 // RUN: %clang_cc1 -triple dxil-pc-shadermodel6.3-library -x hlsl -o - -fsyntax-only %s -verify
 
-// expected-error@+1 {{invalid resource class specifier 'c' used; expected 'b', 's', 't', or 'u'}}
-float a : register(c0, space1);
+// valid, The register keyword in this statement isn't binding a resource, rather it is
+// specifying a constant register binding offset within the $Globals cbuffer, which is legacy behavior from DX9.
+float a : register(c0);
```

Comment:
Do we have a test for static float?
Something like
```
static float sa : register(c1);
```

Reply:
I've added a test case to this file to test this.

Reply:
Got another test idea

```
 groupshared float a[10] : register(c1);
```

Sorry for not getting all the ideas at once :(

Reply:
No worries, added the test case along with the error it produced. Treating groupshared variables as `other`.
Also added some test cases for basic types inside arrays.